lib/streamlit/elements/lib/utils.py

```diff
@@ -148,6 +150,8 @@ def _compute_element_id(
     """
     h = hashlib.new("md5", **HASHLIB_KWARGS)
     h.update(element_type.encode("utf-8"))
+    if user_key:
+        h.update(user_key.encode("utf-8"))
```

Comment:
I guess its added because `key` was previously passed as a `kwarg`, but I am wondering why the `user_key` has to be part of the hash if we append the `user_key` anyways to the final result, which makes the id unique even if there were a collision? It feels like an unnecessary computational step, but maybe there was a good reason for it in the past or still is.

Reply:
Yep indeed, it doesn't have to be added because it's still unique: But I decided to add it since it is closer to the previous behavior and feels like a safer choice in the future in case we change something about the element ID structure. But I don't have a strong preference. Fine with removing this again.